From 6d90552e93a34cf82f2a01e4c45a942e9e4e525d Mon Sep 17 00:00:00 2001
From: Harish Rachakonda <quic_rachakon@quicinc.com>
Date: Fri, 12 Jan 2024 14:41:01 +0530
Subject: [PATCH] hostapd: Fix AID allocation for legacy STA.

Currently, AID is not allocated properly in
hostapd for legacy STA in case of MLD AP. All
the stations have same AID.

Fix this issue by allocating AID properly in
hostapd.

Signed-off-by: Harish Rachakonda <quic_rachakon@quicinc.com>
---
 src/ap/ieee802_11.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index 7339ffe..d26aff0 100644
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -3361,7 +3361,7 @@ static u32 hostapd_get_aid_word(struct hostapd_data *hapd,
 
 	/* Do not assign an AID that is in use on any of the affiliated links
 	 * when finding an AID for a non-AP MLD. */
-	if (hapd->conf->mld_ap) {
+	if (hapd->conf->mld_ap && sta->mld_info.mld_sta) {
 		int j;
 
 		for (j = 0; j < MAX_NUM_MLD_LINKS; j++) {
-- 
2.17.1

